### PR TITLE
Fixing the Windows nested package issues with running Nightwatch and Jest

### PR DESCRIPTION
**Pull request by BenBoersma**
### Summary
Resolves https://github.com/cerner/terra-core/issues/815, this fixes the pathing issues trying to run nightwatch, but nightwatch itself still does not run.
Resolves https://github.com/cerner/terra-core/issues/813

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-props-table/package.json [hunk 1]
```diff
@@ -44,7 +44,6 @@
     "lint:js": "eslint --ext .js,.jsx . --ignore-path ../../.eslintignore",
     "lint:scss": "stylelint src/**/*.scss",
     "test": "npm run test:jest && npm run test:wdio",
-    "test:jest": "jest ./tests/jest/* --config ../../jestconfig.json",
-    "test:wdio": "../../node_modules/.bin/wdio ../../node_modules/terra-dev-site/config/wdio/wdio.conf.js"
-  }
+    "test:jest": "node ../../scripts/run-jest/index.js",
+    "test:wdio": "../../node_modules/.bin/wdio ../../node_modules/terra-dev-site/config/wdio/wdio.conf.js"  }
```
Comment (mjhenkes) [inline packages/terra-props-table/package.json]: move curly brace to next line

File: scripts/run-jest/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const spawn = require('child_process').spawn;
```
Comment (mjhenkes) [inline scripts/run-jest/index.js]: What is this file (add comments)? And if it's common it should live somewhere common so we don't have to duplicate it across repos.

Comment (BenBoersma) [reply scripts/run-jest/index.js]: Its a nodejs module, https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options

**Comment by BenBoersma** [comment]
@mjhenkes 
I just checked and indeed it is having that issue.
Specifically went to "packages/terra-card" and ran "test:wdio: and got 

> '..' is not recognized as an internal or external command,

File: scripts/run-jest/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const os = require('os');
```
Comment (emilyrohrbough) [inline scripts/run-jest/index.js]: I notice we didn't add this as a dependency in our root level package.json

File: scripts/run-jest/index.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  case 'Linux':
+  case 'Darwin':
+    regex = `${process.env.npm_package_name}/tests`;
+    jestcmd = 'jest';
+    break;
+  case 'Windows_NT':
+    regex = `${process.env.npm_package_name}\\\\tests`;
+    jestcmd = 'jest.cmd';
+    break;
+  default:
```
Comment (emilyrohrbough) [inline scripts/run-jest/index.js]: I am worried this check is too specific? What are the possible type returns? could we check for windows and if not windows use `jestcmd= 'jest' by default?

Comment (emilyrohrbough) [reply scripts/run-jest/index.js]: Is the `.cmd` a by product of all bin scripts on windows machines? If so we might look into opening this up to accept a cmd paramater to toggle `jest || jest.cmd` or for example `wdio || wdio.cmd`

Comment (emilyrohrbough) [reply scripts/run-jest/index.js]: Does this command correctly pass command line arguments ? For example, running from terra-button, does this run?

```bash
npm run tt:wdio -- -u
```

**Comment by BenBoersma** [comment]
Changed the way I fixed the issue with Jest. Now instead of creating a custom script to run jest, I have switch jestconfig.json to jestconfig.js and updated its testMatch regex to dynamically match the directory it is called from.

File: jestconfig.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
```
Comment (mjhenkes) [inline jestconfig.js]: Fun Fact: We have two jest configs floating around in our repo. This one is used when running jest in the individual packages, one in package.json is run when running jest on the lerna repo overall?
Two questions,
1. Does running jest at the main repo have an issue on PC?
2. Could you consolidate these configs (I think you could just delete the config from package.json)

rockfact

Comment (BenBoersma) [reply jestconfig.js]: 1. Running Jest on the main repo has no issue on PC.

2. Looking into it now

**Pull request merged**